xapian-applications/omega/postprocess.py

```diff
@@ -0,0 +1,182 @@
+# Postprocess click data files.
+#
+# Copyright (C) 2017 Vivek Pal
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+# USA
+
+"""Postprocesses search and clicks data files.
+
+Generates the final clickstream log file and Query file for Xapian Letor
+module from that log file.
+"""
+
+import argparse
+import collections
+import csv
+import sys
+
+parser = argparse.ArgumentParser(
+    description='''Postprocess click data files.
+
+    This script generates the final clickstream log file from input search and
+    click log files and creates Query file that can be used by Xapian Letor
+    module for generating its training files.
+
+    Expects two log files named "search.log" and "clicks.log" in a directory
+    named "log" as two arguments.
+    ''', formatter_class=argparse.RawTextHelpFormatter)
+parser.add_argument("search_log", type=str, default="log/search.log",
+                    help="Path to the search.log file.")
+parser.add_argument("clicks_log", type=str, default="log/search.log",
+                    help="Path to the clicks.log file.")
+args = parser.parse_args()
+
+
+def generate_final_log(search_log, clicks_log):
+    """Generates Query file formatted as per Xapian Letor documentation.
+
+    Input Args:
+        search_log (str): Path to the search.log file.
+        clicks_log (str): Path to the clicks.log file.
+
+    Example (tab-delimited) entries in search_log:
+
+    821f03288846297c2cf43c34766a38f7	"book"	45,36,14,54,42,52,2,3,15,32	0
+    d41d8cd98f00b204e9800998ecf8427e	""		0
+    d41d8cd98f00b204e9800998ecf8427e	""		0
+    098f6bcd4621d373cade4e832627b4f6	"test"	45,36,14,54,42,52,2,3,15,32	0
+
+    Example (tab-delimited) entries in clicks_log:
+
+    821f03288846297c2cf43c34766a38f7	54
+    821f03288846297c2cf43c34766a38f7	54
+    098f6bcd4621d373cade4e832627b4f6	42
+
+    Example (tab-delimited) entries in final.log:
+
+    QueryID Query   Hits    Offset  Clicks
+    821f03288846297c2cf43c34766a38f7    book    ['45', '36', '14', '54', '42',\
+    '52', '2', '3', '15', '32'] 0   ['45:0', '36:0', '14:0', '54:2', '42:0',\
+    '52:0', '2:0', '3:0', '15:0', '32:0']
```

Comment:
I still don't understand why you've turned a simple and compact representation of the third field into something considerably more unwieldy.